src/ts/content/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { PageContent } from "../common/types";
+import { buildSpecFrom } from "../parsers";
+import { OpenAPIObject } from "openapi3-ts";
+import debug from "../common/logging";
+
+const debugLog = debug("unmock:content:utils");
+
+let apiCheckResult: boolean;
+
+/**
+ * Check if can build a non-trivial OpenAPI spec from the page content. Cache the result to `apiCheckResult` to avoid
+ * building spec everytime this is called. This could mean missing some paths not loaded when this is first executed.
+ */
+export const checkAndCacheApiCheckResult = async (
+  pageContent: PageContent
+): Promise<boolean> => {
+  if (typeof apiCheckResult !== "undefined") {
+    debugLog("Cached API check result", apiCheckResult);
+    return apiCheckResult;
+  }
+  apiCheckResult = await checkIfCanParsePathsFromPage(pageContent);
+  return !!apiCheckResult;
+};
+
+export const checkIfCanParsePathsFromPage = async (
+  pageContent: PageContent
+): Promise<boolean | undefined> => {
+  let isApi: boolean;
+  try {
+    const spec: OpenAPIObject = await buildSpecFrom(pageContent);
+    isApi = Object.keys(spec.paths).length > 0;
+  } catch (err) {
+    console.error("Failed building OpenAPI spec from page", err);
```

Comment:
`return false` here so this function strictly returns a `Promise<boolean>` instead of multiple types

Reply:
Could do, but then it needs some kind of retry logic in case something goes wrong in the first time. Enabling strict null checks is something some future warrior will have to do at some point probably.

Reply:
Added `false` by default, it's only tried once then.